### PR TITLE
ci: 👷 build and push workflow

### LINKED ISSUE #7: Build and push on merge
The changes that get merged onto develop branch are usually required for dependent PRs in the main repo. The `image:dev` should be pushed so that changes are in sync and the main application works without any issue.

### PR DESCRIPTION
**Pull request by prashant-shahi**
Fixes #7 

Signed-off-by: Prashant Shahi <prashant@signoz.io>

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+on:
+  push:
+    branches:
+      - main
+      - develop
+    tags:
+      - v*
+
+jobs:
+  image-build-and-push-query-service:
```
Comment (srikanthccv) [inline .github/workflows/push.yaml:12]:
```suggestion
  image-build-and-push-signoz-collector:
```

File: Makefile [hunk 2]
```diff
@@ -42,13 +42,13 @@ fmt:
 	@echo Running go fmt on query service ...
 	@$(GOFMT) -e -s -l -w .
 
-.PHONY: build-signoz-collector
-build-signoz-collector:
+.PHONY: build-and-push-signoz-collector
+build-and-push-signoz-collector:
```
Comment (srikanthccv) [inline Makefile:46]: Can you add a new target to this build-and-push? When working locally it helps to have a separate target which just creates a new image.

**Review by makeavish — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (prashant-shahi) [reply Makefile:46]: @srikanthccv Done.

File: .github/workflows/push.yaml (hunk 1 above)
Comment (prashant-shahi) [reply .github/workflows/push.yaml]: Done.

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**